### PR TITLE
TIAF Native Console

### PR DESCRIPTION
**Pull request by jonawals**
Signed-off-by: John <jonawals@amazon.com>

## What does this PR do?

This PR is a series of PRs that implement the necessary changes for TIAF to support the native (C++) tests with the common library code from [PR-429](https://github.com/aws-lumberyard-dev/o3de/pull/429). This PR in particular moves all of the native console functionality into dedicated build targets that make use of the aforementioned common library build targets.

**Note:** This PR contains a lot of previously reviewed files. These have been annotated accordingly and can be ignored for the purposes of this review. Furthermore, this PR is going into a staging buffer that may require additional review once this PR has been merged.

### CODE REVIEW

File: Code/Tools/TestImpactFramework/Frontend/Console/Common/Code/Include/Headers/TestImpactFramework/TestImpactConsoleMain.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Frontend/Console/Common/Code/Include/Headers/TestImpactFramework/TestImpactConsoleMain.h:1]: This file has been previously reviewed (pre Github).

File: Code/Tools/TestImpactFramework/Frontend/Console/Native/Code/Source/TestImpactConsoleMain.cpp [hunk 2]
```diff
@@ -0,0 +1,278 @@
+/*
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Frontend/Console/Native/Code/Source/TestImpactConsoleMain.cpp:1]: This file has been previously reviewed (pre Github).

File: Code/Tools/TestImpactFramework/Frontend/Console/CMakeLists.txt [hunk 3]
```diff
@@ -6,4 +6,5 @@
 #
 #
 
-add_subdirectory(Code)
\ No newline at end of file
+add_subdirectory(Common)
+add_subdirectory(Native)
```
Comment (hultonha) [inline Code/Tools/TestImpactFramework/Frontend/Console/CMakeLists.txt:10]: nit: Missing EOL character

File: Code/Tools/TestImpactFramework/Frontend/Console/Native/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+add_subdirectory(Code)
```
Comment (hultonha) [inline Code/Tools/TestImpactFramework/Frontend/Console/Native/CMakeLists.txt:9]: nit: EOL character

File: Code/Tools/TestImpactFramework/Frontend/Console/Native/Code/Source/TestImpactCommandLineOptions.cpp [hunk 5]
```diff
@@ -430,6 +463,8 @@ namespace TestImpact
             "                                                                <tiaf binay build dir>.<tiaf binary build type>.json).\n"
             "    -datafile=<filename>                                        Optional path to a test impact data file that will used instead of that\n"
             "                                                                specified in the config file.\n"
+            "    -previousrundatafile=<filename>                             Optional path to a test impact data file that will used instead of that\n"
```
Comment (hultonha) [inline Code/Tools/TestImpactFramework/Frontend/Console/Native/Code/Source/TestImpactCommandLineOptions.cpp:466]: nit: will used -> will be used

**Review by hultonha — APPROVED** [review]
Overall looks good :+1:

File: Code/Tools/TestImpactFramework/Frontend/Console/CMakeLists.txt (hunk 3 above)
Comment (jonawals) [reply Code/Tools/TestImpactFramework/Frontend/Console/CMakeLists.txt:10]: Fixed 👍

File: Code/Tools/TestImpactFramework/Frontend/Console/Native/CMakeLists.txt (hunk 4 above)
Comment (jonawals) [reply Code/Tools/TestImpactFramework/Frontend/Console/Native/CMakeLists.txt:9]: Fixed 👍

File: Code/Tools/TestImpactFramework/Frontend/Console/Native/Code/Source/TestImpactCommandLineOptions.cpp (hunk 5 above)
Comment (jonawals) [reply Code/Tools/TestImpactFramework/Frontend/Console/Native/Code/Source/TestImpactCommandLineOptions.cpp:466]: Fixed 👍

**Review by amzn-jillich — APPROVED** [review]

**Pull request merged**